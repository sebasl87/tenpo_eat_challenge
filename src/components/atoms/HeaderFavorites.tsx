import React from 'react'
import styled from 'styled-components/native';

export interface IHeaderFavorites {
    imageLogo?: any;
    imageBg?: any;

}

const LayoutImage = styled.ImageBackground`
    width: 260px;
    height: 102px;
`
const LogoImage = styled.Image`
    width: 44px;
    height: 44px;    
    position: absolute;
    z-index: 1;
    margin-left: 11px;
    margin-top: 9px;

`

const Container = styled.View`
    width: 260px;
    height: 102px;
    display: flex;            
`

const HeaderFavorites: React.FC<IHeaderFavorites> = ({ imageLogo, imageBg }) => {
    return (
        <Container>
            <LogoImage source={imageLogo} />
            <LayoutImage source={imageBg}>
            </LayoutImage>
        </Container>

    )
}

export default HeaderFavorites;