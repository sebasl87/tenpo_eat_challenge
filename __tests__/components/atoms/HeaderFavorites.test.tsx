import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import HeaderFavorites from "../../../src/components/atoms/HeaderFavorites";


describe("HeaderFavorites component", () => {
    it("should render successfully with HeaderFavorites", () => {
        const titleSection = renderer
            .create(<HeaderFavorites />)
            .toJSON();

        expect(titleSection).toMatchSnapshot();
    });
});
