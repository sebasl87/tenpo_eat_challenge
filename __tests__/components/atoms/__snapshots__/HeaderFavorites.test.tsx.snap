// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFavorites component should render successfully with HeaderFavorites 1`] = `
<View
  style={
    [
      {
        "display": "flex",
        "height": 102,
        "width": 260,
      },
    ]
  }
>
  <Image
    style={
      [
        {
          "height": 44,
          "marginLeft": 11,
          "marginTop": 9,
          "position": "absolute",
          "width": 44,
          "zIndex": 1,
        },
      ]
    }
  />
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      [
        {
          "height": 102,
          "width": 260,
        },
      ]
    }
  >
    <Image
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 102,
            "width": 260,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
